### ISSUE TITLE
Project Level 3

### ISSUE THREAD

**Issue by loqir** [nus-cs2030/2122-s2#161]
is this how this level to supposed to work?

i create classes for all the events,

in the main3 class,
for each input, i create a new Arrive event with a customer inside, and call execute until there is no event, while at each step i add the event into the priority queue.

I then pq.poll() until the priority queue is empty?

If this is indeed the case, are the flow of events specific to a customer? so the Done event will return empty instead of Serve if there is another customer waiting?

Also, how do I get the arrival time of the previous customer if one customer is currently waiting? Am i supposed to have a list of customers for each server?

**Comment by sheryew** [comment]
Yes, you will poll till your priority queue is empty. 

My Done event returned an empty event. 

I created the variable named availableTime in my server class that stores the time the server will be free. 

Not too sure whether that's the right approach but it worked for me

**Comment by mazchew** [comment]
for my implementation, the flow of events was specific to a customer, i think that's a much simpler solution that having the flow of events being shared by all customers.

yup, i had a list of customers for each server

**Comment by LohYangXian** [comment]
Hi, I returned an empty optional for my Done event's execute() as well. 

For my simulate3, I created new Arrive Events for all customers and tried to execute the events until there are no more events left in the Queue.
e.g. if the execute returns an event (e.g. SERVE), I would add it to PriorityQueue and update the Shop.
 if execute returns a Optional.empty() (like from my DONE event), i would only update the shop and not add any event back to PriorityQueue. Hope this helps!
